js/src/ui/Form/AddressSelect/addressSelect.js

```diff
@@ -36,31 +35,75 @@ export default class AddressSelect extends Component {
     onChange: PropTypes.func.isRequired
   }
 
+  state = {
+    entries: {}
+  }
+
+  componentWillMount () {
+    const { accounts, contacts } = this.props;
+    const entries = Object.assign({}, accounts || {}, contacts || {});
+    this.setState({ entries });
+  }
+
+  componentWillReceiveProps (newProps) {
+    const { accounts, contacts } = newProps;
+    const entries = Object.assign({}, accounts || {}, contacts || {});
+    this.setState({ entries });
+  }
+
   render () {
-    const { disabled, error, hint, label, value } = this.props;
+    const { disabled, error, hint, label } = this.props;
 
     return (
-      <Select
-        disabled={ disabled }
-        label={ label }
-        hint={ hint }
-        error={ error }
-        value={ value }
-        onChange={ this.onChange }>
-        { this.renderSelectEntries() }
-      </Select>
+      <div>
+        <AutoComplete
```

Comment:
Could we (for the sake of consistency in the input components), add the AutoComplete to where we have the other input components, i.e. a thin wrapper. It feels like the odd man out atm.